### DISCUSSION TITLE
Deploy de Typescript/Backend com tsnode ou dos arquivos buildados?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1661] [Perguntas e Respostas]
Fala galera, 
atualmente atuo com node+ express ("puros") no backend, mas fiquei encantado com o TypeScript e vou adotá-lo full em meus projetos backends (futuramente após acostumar passo para o front também).

Mas enfim, a dúvida: Aqui em desenvolvimento eu rodo o tsnode, que pelo que vi ele já roda direto o .ts... Para fazer o deploy em produção eu uso o PM2 + tsnode e rodo direto o .ts? Ou eu faço deploy dos arquivos buildados e rodo com o PM2 + Node mesmo igual faço com meus scripts "puros"?

Muito obrigado!
Abraços!

**Comment by vhfmag** [comment]
A documentação do [`ts-node`](https://npm.im/ts-node) não fala nada sobre, mas a do [`@babel/node`](https://babeljs.io/docs/en/next/babel-node.html) diz o seguinte:

> **Not meant for production use**
> You should not be using babel-node in production. It is unnecessarily heavy, with high memory usage due to the cache being stored in memory. You will also always experience a startup performance penalty as the entire app needs to be compiled on the fly.

Eu imagino que os mesmos problemas se apliquem ao `ts-node`... Tirando isso, recomendaria de usar Babel pra transpilar Typescript (e Typescript pra validar os tipos), pela flexibilidade de poder usar outros plugins e presets :)
